community/detectors/apache_activemq_cve_2023_46604/src/test/java/com/google/tsunami/plugins/detectors/cves/cve202346604/Cve202346604DetectorTest.java

```diff
@@ -0,0 +1,214 @@
+                .build(),
+            new AbstractModule() {
+              @Override
+              protected void configure() {
+                OptionalBinder.newOptionalBinder(
+                        binder(), Key.get(SocketFactory.class, SocketFactoryInstance.class))
+                    .setBinding()
+                    .toInstance(socketFactoryMock);
+              }
+            },
+            new HttpClientModule.Builder().build())
+        .injectMembers(this);
+  }
+
+  @Test
+  public void detect_whenVulnerable_returnsVulnerability() throws Exception {
+    final byte[] serverInfoResponse =
+        new byte[] {
+          0, 0, 1, 82, 1, 65, 99, 116, 105, 118, 101, 77, 81, 0, 0, 0, 12, 1, 0, 0, 1, 64, 0, 0, 0,
+          13, 0, 17, 83, 116, 97, 99, 107, 84, 114, 97, 99, 101, 69, 110, 97, 98, 108, 101, 100, 1,
+          1, 0, 15, 80, 108, 97, 116, 102, 111, 114, 109, 68, 101, 116, 97, 105, 108, 115, 9, 0, 4,
+          74, 97, 118, 97, 0, 12, 67, 97, 99, 104, 101, 69, 110, 97, 98, 108, 101, 100, 1, 1, 0, 17,
+          84, 99, 112, 78, 111, 68, 101, 108, 97, 121, 69, 110, 97, 98, 108, 101, 100, 1, 1, 0, 18,
+          83, 105, 122, 101, 80, 114, 101, 102, 105, 120, 68, 105, 115, 97, 98, 108, 101, 100, 1, 0,
+          0, 9, 67, 97, 99, 104, 101, 83, 105, 122, 101, 5, 0, 0, 4, 0, 0, 12, 80, 114, 111, 118,
+          105, 100, 101, 114, 78, 97, 109, 101, 9, 0, 8, 65, 99, 116, 105, 118, 101, 77, 81, 0, 20,
+          84, 105, 103, 104, 116, 69, 110, 99, 111, 100, 105, 110, 103, 69, 110, 97, 98, 108, 101,
+          100, 1, 1, 0, 12, 77, 97, 120, 70, 114, 97, 109, 101, 83, 105, 122, 101, 6, 0, 0, 0, 0, 6,
+          64, 0, 0, 0, 21, 77, 97, 120, 73, 110, 97, 99, 116, 105, 118, 105, 116, 121, 68, 117, 114,
+          97, 116, 105, 111, 110, 6, 0, 0, 0, 0, 0, 0, 117, 48, 0, 32, 77, 97, 120, 73, 110, 97, 99,
+          116, 105, 118, 105, 116, 121, 68, 117, 114, 97, 116, 105, 111, 110, 73, 110, 105, 116, 97,
+          108, 68, 101, 108, 97, 121, 6, 0, 0, 0, 0, 0, 0, 39, 16, 0, 19, 77, 97, 120, 70, 114, 97,
+          109, 101, 83, 105, 122, 101, 69, 110, 97, 98, 108, 101, 100, 1, 1, 0, 15, 80, 114, 111,
+          118, 105, 100, 101, 114, 86, 101, 114, 115, 105, 111, 110, 9, 0, 6, 53, 46, 49, 55, 46, 51
+        };
+
+    configureMockSocket(new String(serverInfoResponse, StandardCharsets.UTF_8));
+    NetworkService service =
+        NetworkService.newBuilder()
+            .setNetworkEndpoint(forIpAndPort("127.0.0.1", 1234))
+            .setTransportProtocol(TransportProtocol.TCP)
+            .setSoftware(Software.newBuilder().setName("ActiveMQ"))
+            .build();
+    TargetInfo targetInfo = TargetInfo.getDefaultInstance();
+    PollingResult log = PollingResult.newBuilder().setHasHttpInteraction(true).build();
+    String body = JsonFormat.printer().preservingProtoFieldNames().print(log);
+    mockCallbackServer.enqueue(
+        new MockResponse().setResponseCode(HttpStatus.OK.code()).setBody(body));
```

Comment:
These lines can be replaced with: `mockCallbackServer.enqueue(PayloadTestHelper.generateMockSuccessfulCallbackResponse());`

Reply:
Okey